lib/slack/block_kit/utils.rb

```diff
@@ -0,0 +1,23 @@
+module Slack
+  module BlockKit
+    module Utils
+      # Truncate a string to a certain length, adding an optional omission
+      # string if the string is truncated. This is generally used to ensure
+      # that certain block limits do not exceed their maximum length when
+      # autofixing is enabled.
+      #
+      # See: https://api.rubyonrails.org/classes/String.html#method-i-truncate
+      def self.truncate(text, length:, omission: "...", separator: nil)
+        return text.dup unless text.length > length
+
+        omission ||= ""
+        length_with_room_for_omission = length - omission.length
+
+        stop = text.rindex(separator, length_with_room_for_omission) if separator
+        stop ||= length_with_room_for_omission
+
+        "#{text[0, stop]}#{omission}"
+      end
```

Comment:
Note: I used Rails' implementation of `truncate` for a couple reasons

1. Certain things should maybe truncate differently (e.g. `block_id` and `action_id` don't need to truncate with an ellipsis at the end, though that's supposedly still valid), so the `omission` and `separator` options give us control
2. I could see us adding configuration options on top of `autofix_invalid_blocks` that allow users to specify how they'd like text to truncate